### PR TITLE
feat(ui): global confirm modal

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Introduce global `ConfirmModalProvider` to handle all the confirm modal actions within the page.
Add `useConfirmModal` hook to interact with the global confirm modal context provider. The show modal method will return a promise to promisify a modal confirm flow. 

Show the alert modal at the passcode validation page, if no target email/sms address is found. 

image

Will refactor all current passwordlessModal and TermsOfUseModal using the new implemented global `ConfirmModalProvider`.

## Testing

test locally
ut added

### CODE REVIEW

File: packages/ui/src/components/ConfirmModal/AcModal.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ import { ModalProps } from './type';
 const AcModal = ({
```
Comment (charIeszhao) [inline packages/ui/src/components/ConfirmModal/AcModal.tsx:13]: I know this is old code now, but I'm wondering why do we call it `AcModal`? What does it mean?

File: packages/ui/src/components/ConfirmModal/AcModal.tsx [hunk 2]
```diff
@@ -37,8 +38,23 @@ const AcModal = ({
...
+              onClose();
+            }}
+          />
+          {type === 'confirm' && (
+            <Button
+              title={confirmText}
+              size="small"
+              onClick={() => {
+                (onConfirm ?? onClose)();
+              }}
```
Comment (charIeszhao) [inline packages/ui/src/components/ConfirmModal/AcModal.tsx:55]: Why do we change the usage of `onClick` handler? Seems the old pattern is more efficient

File: packages/ui/src/components/ConfirmModal/AcModal.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/AcModal.tsx:13]: hmm, it's for desktop only and migrated from AC 😆

File: packages/ui/src/components/ConfirmModal/AcModal.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/AcModal.tsx:55]: avoid the default onClick event being passed to the callback method. Since asked,  Let me check if there is a more proper way.

File: packages/ui/src/components/ConfirmModal/MobileModal.tsx [hunk 3]
```diff
@@ -29,8 +27,21 @@ const MobileModal = ({
...
-          <Button title={cancelText} type="secondary" onClick={onClose} />
-          <Button title={confirmText} onClick={onConfirm ?? onClose} />
+          <Button
+            title={cancelText}
+            type="secondary"
+            onClick={() => {
+              onClose();
+            }}
+          />
+          {type === 'confirm' && (
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/MobileModal.tsx:37]: make sense

File: packages/ui/src/components/ConfirmModal/AcModal.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/AcModal.tsx:55]: changed the handle logic to the upper scope ConfirmModalProvider

File: packages/ui/src/components/ConfirmModal/MobileModal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/components/ConfirmModal/MobileModal.tsx]: updated @charIeszhao please check

**Review by charIeszhao — APPROVED** [review]

File: packages/ui/src/containers/TermsOfUse/TermsOfUseConfirmModal/index.tsx [hunk 4]
```diff
@@ -47,7 +47,9 @@ const TermsOfUseConfirmModal = ({ isOpen = false, onConfirm, onClose }: Props) =
         setTermsAgreement(true);
         onConfirm();
       }}
-      onClose={onClose}
+      onClose={() => {
+        onClose();
+      }}
```
Comment (charIeszhao) [inline packages/ui/src/containers/TermsOfUse/TermsOfUseConfirmModal/index.tsx:52]: Maybe we can also revert this change

Comment (simeng-li) [reply packages/ui/src/containers/TermsOfUse/TermsOfUseConfirmModal/index.tsx:52]: Unfortunately not, as we keep the onClose as well as onConfirm method as the original MouseEventHandler. 
But the TermsOfUseConfirmModal onClose props type was (message?: string) => void;  Have to do the trick.
